src/clusterfuzz/_internal/bot/tasks/utasks/progression_task.py

```diff
@@ -222,43 +345,56 @@ def _store_testcase_for_regression_testing(testcase, testcase_file_path):
                    regression_testcase_url)
 
 
-def find_fixed_range(testcase_id, job_type):
-  """Attempt to find the revision range where a testcase was fixed."""
-  deadline = tasks.get_task_completion_deadline()
+def utask_preprocess(testcase_id, job_type, uworker_env):
+  """Runs preprocessing for progression task."""
   testcase = data_handler.get_testcase_by_id(testcase_id)
   if not testcase:
-    return
+    return None
 
   if testcase.fixed:
-    logs.log_error('Fixed range is already set as %s, skip.' % testcase.fixed)
-    return
-
-  # Setup testcase and its dependencies.
-  _, testcase_file_path, error = setup.setup_testcase(testcase, job_type)
-  if error:
-    # TODO(https://github.com/google/clusterfuzz/issues/3008): Change this when
-    # progression is migrated.
-    all_errors = uworker_handle_errors.get_all_handled_errors()
-    uworker_handle_errors.handle(error, all_errors)
-    return
+    logs.log_error(f'Fixed range is already set as {testcase.fixed}, skip.')
+    return None
 
+  # TODO(alhijazi): Make sure this is always properly cleared on failure.
   # Set a flag to indicate we are running progression task. This shows pending
   # status on testcase report page and avoid conflicting testcase updates by
   # triage cron.
   testcase.set_metadata('progression_pending', True)
+  data_handler.update_testcase_comment(testcase, data_types.TaskState.STARTED)
+
+  return uworker_io.UworkerInput(
+      job_type=job_type,
+      testcase_id=testcase_id,
+      uworker_env=uworker_env,
+      testcase=testcase,
+  )
+
+
+def find_fixed_range(uworker_input):
+  """Attempt to find the revision range where a testcase was fixed."""
+  deadline = tasks.get_task_completion_deadline()
+  testcase_id = uworker_input.testcase_id
+  testcase = uworker_input.testcase
+  job_type = uworker_input.job_type
+
+  # Setup testcase and its dependencies.
+  _, testcase_file_path, error = setup.setup_testcase(testcase, job_type)
+  if error:
+    return error
 
   # Custom binaries are handled as special cases.
   if build_manager.is_custom_binary():
-    _check_fixed_for_custom_binary(testcase, job_type, testcase_file_path)
-    return
+    return _check_fixed_for_custom_binary(uworker_input, testcase,
+                                          testcase_file_path)
 
   build_bucket_path = build_manager.get_primary_bucket_path()
   revision_list = build_manager.get_revisions_list(
       build_bucket_path, testcase=testcase)
   if not revision_list:
-    data_handler.close_testcase_with_error(testcase,
-                                           'Failed to fetch revision list')
-    return
+    return uworker_io.UworkerOutput(
+        testcase=testcase,
+        uworker_input=uworker_input,
```

Comment:
so what you're saying is, I only need to set `uworker_input` when I pass to other non-trusted modules right? This means that for all error handling reporting stuff, we don't need to set `uworker_input`.

The reason I thought it's necessary is that we are creating `uworker_input` manually in `setup_testcase` and returning it to the trusted worker on failure.

Reply:
> so what you're saying is, I only need to set `uworker_input` when I pass to other non-trusted modules right? 

Maybe, It's only useful when the caller isn't migrated yet.
The correct behavior when you get a uworker_output containing an error is to return from main and let postprocess handle it. In that case we don't need to add uworker_input. Because when we download uworker_output, we also download uworker_input. However, for unmigrated users that aren't yet behaving properly, we need to include uworker_input.

>This means that for all error handling reporting stuff, we don't need to set `uworker_input`.

Yes, if the caller is migrated properly, then we don't need to set it.

> 
> The reason I thought it's necessary is that we are creating `uworker_input` manually in `setup_testcase` and returning it to the trusted worker on failure.

Right, setup_testcase has many uunmigrated callers which is why we need this.

Reply:
`uworker_input` not set anymore in `uworker_output` from utask_main